agent/api/task.go

```diff
@@ -202,23 +208,64 @@ func (task *Task) initializeCredentialsEndpoint(credentialsManager credentials.M
 
 }
 
+// BuildCNIConfig constructs the cni configuration from eni
+func (task *Task) BuildCNIConfig() (*ecscni.Config, error) {
+	if !task.isNetworkModeVPC() {
+		return nil, errors.New("task isnot enabled with vpc networkmode")
+	}
+
+	cfg := &ecscni.Config{}
+	eni := task.GetTaskENI()
+	if eni == nil {
+		return nil, errors.New("task has no eni associated")
+	}
+
+	cfg.ENIID = eni.ID
+	cfg.ID = eni.MacAddress
+	cfg.ENIMACAddress = eni.MacAddress
+	for _, ipv4 := range eni.IPV4Addresses {
+		if ipv4.Primary {
+			cfg.ENIIPV4Address = ipv4.Address
+			break
+		}
+	}
+
+	// If there is ipv6 assigned to eni then set it
+	if len(eni.IPV6Addresses) == 1 {
```

Comment:
Can you modify this to be `> 0` instead of `== 1`? `> 0` is not validating anything, where as `== 1` is. Here, we do not want to validate anything as we assume that everything has been validated for us to reach here.